superset-frontend/packages/superset-ui-chart-controls/src/operators/resampleOperator.ts

```diff
@@ -28,13 +32,21 @@ export const resampleOperator: PostProcessingFactory<
   const resampleMethod = resampleZeroFill ? 'asfreq' : formData.resample_method;
   const resampleRule = formData.resample_rule;
   if (resampleMethod && resampleRule) {
+    const groupby_columns = ensureIsArray(queryObject.columns).map(column => {
+      if (isPhysicalColumn(column)) {
+        return column;
+      }
+      return column.label;
+    });
```

Comment:
There's already a `getColumnLabel` util for this:
```javascript
const groupby_columns = ensureIsArray(queryObject.columns).map(column => getColumnLabel);
```

Reply:
I think the `sqlExpression` can't fit in `pandas` column, so I deliberately let it produce `undefine`.

```
....
export default function getColumnLabel(column: QueryFormColumn): string {
  if (isPhysicalColumn(column)) {
    return column;
  }
  if (column.label) {
    return column.label;
  }
  return column.sqlExpression;
}
```

Reply:
@zhaoyongjie maybe we should change the label to an MD5 or some other hash if `column.label` is falsy? I think for the most part we don't need the label to be equal to `sqlExpression`, but rather to just make sure it's deterministic.

Reply:
@zhaoyongjie btw I just noticed that when creating two adhoc columns with _different_ expressions where the label is _unchanged_, the query only includes the first one:
![image](https://user-images.githubusercontent.com/33317356/149516917-31ecd51b-ec44-4d1f-9291-7114fe88ec86.png)
So we should perhaps assign the label in the popover, even just giving them a unique serial id, like "My Column", "My Column (1)", ..., "My Column (n)".

Ping @kgabryje

Reply:
I like the idea, it shouldn't be difficult to implement. Also, what do you think about creating a validator that returns an error when 2 or more columns or metrics have the same label?

Reply:
> I like the idea, it shouldn't be difficult to implement. Also, what do you think about creating a validator that returns an error when 2 or more columns or metrics have the same label?

Yeah! That would make it bullet proof 👍

Reply:
> @zhaoyongjie maybe we should change the label to an MD5 or some other hash if `column.label` is falsy? I think for the most part we don't need the label to be equal to `sqlExpression`, but rather to just make sure it's deterministic.

IMO, we have to give every adhoc column a explicit and unique name. This name not only used in `alias` in the sql, but also in pandas column.

Reply:
@zhaoyongjie I'm ok with this, but I'd recommend following up on this quickly, as we're now in a kind-of bad place with stacked tech debt due to this:
- possibility of duplicate column labels
- needed workaround due to `getColumnLabel` not being universally applicable due to the label property being optional
- the type `groupby_columns?: Array<string | undefined>;` which IMO should just be `groupby_columns?: string[]`

Reply:
I have created a Shotcut ticket to record this.